### PR TITLE
VideoPress: Return a highly compatible video file format for non-VideoPress players

### PR DESCRIPTION
**Pull request by jgcaruso**
Fixes #

Screen Shot 2022-02-03 at 5 18 27 PM
Screen Shot 2022-02-03 at 5 27 34 PM

This PR introduces support for the new "compatibility" video format being introduced to new VideoPress uploads. This video file allows non-VideoPress players to display a video file that will be compatible in most (well all modern browsers) instead of the originally uploaded video file (which could be an `.mov` file which wouldn't play in Firefox for example).

This essentially returns functionality for 3rd party players back to the way it was prior to 10.2 and the introduction of HLS streaming in VideoPress, but with a higher resolution (1080p vs 720p before) so there shouldn't be any more compatibility issues with these players, and the playback should look better for customers uploading higher resolution videos.

#### Changes proposed in this Pull Request:

* If available, the "compat" video format is returned to themes, plugins and blocks that call `wp_get_attachment_url()`
* If "compat" is available, a new "Original File URL" field is displayed in the Media Library details window so the original url can be copied and shared if necessary.

#### Does this pull request change what data or activity we track or use?

nope

#### Testing instructions:

* ~~Apply D74319-code to your sandbox if it hasn't been deployed yet~~ deployed!
* Upload a video to the Media Library
* Create a post with a `/cover` block and `/story` block and publish it
* The public page's video urls should point to the `COMPAT` video format (ex `<filename>_hd_1080p_compat.mp4`)
* View the attachment in the Media Library
* The new "Original File Url" filed should be visible with the original file and the "File Url" field should have the COMPAT format
* Add a header video via your theme (if supported)
* View the public page, it should be using the `COMPAT` format

### CODE REVIEW

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 1]
```diff
@@ -192,6 +193,16 @@ public function fields_to_edit( $fields, $post ) {
 			'show_in_edit'  => false,
 		);
 
+		if ( $file_statuses['hd_1080p_compat'] ) {
+			$fields['videopress_original_file_url'] = array(
+				'label'         => _x( 'Original File URL', 'A header for the original file url field', 'jetpack' ),
+				'input'         => 'html',
+				'html'          => "<input type=\"text\" name=\"videopress_original_file_url\" value=\"{$original_file_url}\" readonly=\"readonly\"/>",
```
Comment (thedebian) [inline projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:200]: Should $original_file_url be escaped ?

**Review by thedebian — CHANGES_REQUESTED** [review]
Worked fine for the cover block, but the story block is using the original URL instead of the compat (I see "compat" in the poster image though) : 

Cover block : 
Capture d’écran 2022-02-10 à 09 58 48

Story block : 
Capture d’écran 2022-02-10 à 09 58 33

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php (hunk 1 above)
Comment (jgcaruso) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:200]: The url gets sanitized when we process the file, but probably doesn't hurt to escape it here just to be completely certain.

**Comment by jgcaruso** [comment]
>  but the story block is using the original URL instead of the compat

Looks like the Story block is actually pulling out specific files from the `videopress` object, so it'll require a change to its code.

**Comment by jgcaruso** [comment]
> Worked fine for the cover block, but the story block is using the original URL instead of the compat (I see "compat" in the poster image though) :
> 
> Cover block : Capture d’écran 2022-02-10 à 09 58 48
> 
> Story block : Capture d’écran 2022-02-10 à 09 58 33

So it looks like the file url gets persisted the moment it is selected. So you would need to create a new `/story` block OR re-select the video in the existing `/story` block in order for it to return the COMPAT format.

Here is a screenshot of my inspector, showing the compat file is being used.
Screen Shot 2022-02-17 at 11 06 28 AM

**Review by thedebian — DISMISSED** [review]
Working fine ! :shipit:

**Review by thedebian — CHANGES_REQUESTED** [review]
It worked fine previously, and now, I have the issue with the /story block again.
I tried recreating the block, reselecting the video, but I still don't have the compat format (while the cover block works fine for the same video) : 
Capture d’écran 2022-02-25 à 10 10 01

**Comment by jgcaruso** [comment]
> It worked fine previously, and now, I have the issue with the /story block again. I tried recreating the block, reselecting the video, but I still don't have the compat format (while the cover block works fine for the same video) : Capture d’écran 2022-02-25 à 10 10 01

🤔 Does it work with a completely brand new video?

I wonder if there is some internal caching or metadata that is messing it up?

**Review by thedebian — APPROVED** [review]
After applying a patch to the story block enrich_video_meta function, we've been able to fix the issue. Ready to go ! 🚀

**Comment by jgcaruso** [comment]
PR on hold since we've disabled the `compat` format until we get video job server load under control.

**Comment by jgcaruso** [comment]
Abandoning for reasons given in pbFulr-Cp-p2

**Pull request closed without merging**